test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -1014,14 +1014,21 @@ private void wipeCluster() throws Exception {
 
     private void waitForClusterUpdates() throws Exception {
         logger.info("Waiting for all cluster updates up to this moment to be processed");
+
         try {
             assertOK(adminClient().performRequest(new Request("GET", "_cluster/health?wait_for_events=languid")));
         } catch (ResponseException e) {
             if (e.getResponse().getStatusLine().getStatusCode() == HttpStatus.SC_REQUEST_TIMEOUT) {
+                StringBuilder logMessage = new StringBuilder("Timed out waiting for cluster updates to be processed.");
                 final var pendingTasks = getPendingClusterStateTasks();
                 if (pendingTasks != null) {
-                    logger.error("Timed out waiting for cluster updates to be processed, {}", pendingTasks);
+                    logMessage.append('\n').append(pendingTasks);
+                }
+                final var hotThreads = getHotThreads();
+                if (hotThreads != null) {
+                    logMessage.append("\nHot threads: ").append(hotThreads);
                 }
```

Comment:
Should this be inside the `if (pendingTasks != null)` block?

Reply:
I intentionally didn't put it inside that `if` statement. I figured that if the pending tasks is `null` (which would rarely happen), the hot threads might still be valuable. In any case, I don't think it matters a lot because I don't expect `pendingTasks` to be `null` very often.